code/Backend/flask-backend/app.py

```diff
@@ -380,71 +386,58 @@ def get_reservation():
+def execute_database_query(query, params=None):
+    try:
+        # Connect to the PostgreSQL database
         conn = psycopg2.connect(**db_connection_settings)
         cursor = conn.cursor()
-
-        query = "SELECT * FROM Equipment WHERE name ILIKE %s OR description ILIKE %s"
-        cursor.execute(query, (f"%{search_query}%", f"%{search_query}%"))
-
+        if params is not None:
+            cursor.execute(query, params)
+        else:
+            cursor.execute(query)
+        # Fetch all rows and store them in a list of dictionaries
         columns = [desc[0] for desc in cursor.description]
-        equipment_data = [dict(zip(columns, row)) for row in cursor.fetchall()]
-
+        data = [dict(zip(columns, row)) for row in cursor.fetchall()]
         cursor.close()
         conn.close()
-
-        if equipment_data:
-            return jsonify(equipment_data), 200
-        else:
-            return jsonify({"message": "No matching items found"}), 404
-
+        return data
     except Exception as e:
-        return jsonify({"error": str(e)}), 500
+        return None, str(e)
 
-@app.route('/api/availableItems', methods=['GET'])
-def get_available_items():
+@app.route('/api/searchItems', methods=['GET'])
+def search_items():
     try:
-        conn = psycopg2.connect(**db_connection_settings)
-        cursor = conn.cursor()
-
-        cursor.execute("SELECT * FROM Equipment WHERE available = true")
-        columns = [desc[0] for desc in cursor.description]
-        equipment_data = [dict(zip(columns, row)) for row in cursor.fetchall()]
-
-        cursor.close()
-        conn.close()
 
+        search_query = request.args.get('q')
+        print(f"Received search query: {search_query}")
+        query = "SELECT * FROM Equipment WHERE name ILIKE %s OR description ILIKE %s"
+        params = (f"%{search_query}%", f"%{search_query}%")
+        equipment_data = execute_database_query(query, params)
         if equipment_data:
             return jsonify(equipment_data), 200
         else:
-            return jsonify({"message": "No available items found"}), 404
-
+            return jsonify({"message": "No matching items found"}), 404
     except Exception as e:
         return jsonify({"error": str(e)}), 500
 
-@app.route('/api/unavailableItems', methods=['GET'])
-def get_unavailable_items():
+    
+@app.route('/api/items', methods=['GET'])
+def get_items():
     try:
-        conn = psycopg2.connect(**db_connection_settings)
-        cursor = conn.cursor()
-
-        cursor.execute("SELECT * FROM Equipment WHERE available = false")
-        columns = [desc[0] for desc in cursor.description]
-        equipment_data = [dict(zip(columns, row)) for row in cursor.fetchall()]
-
-        cursor.close()
-        conn.close()
-
+        availability = request.args.get('availability')
+        query = "SELECT * FROM Equipment"
+        if availability == "available":
+            query += " WHERE available = true"
```

Comment:
Same question here: why create another query variable when you have created constants on line 35-36?

Reply:
Same as my last comment.